libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -52,24 +52,25 @@ ENTRY(.rtld_start)
 #ifdef __CHERI_PURE_CAPABILITY__
 	.cfi_undefined	c30
 	mov	c19, c0				/* Put aux in a callee-saved register */
-#ifdef RTLD_SANDBOX
```

Comment:
it's not obvious to me why the changes in rtld_start.S are needed in this commit.  (Also, I would move this one commit to a separate review as well perhaps so it can be merged earlier)

Reply:
This is needed because now `c18n_init_rtld_stack` must be called after RTLD relocates itself. Since it now references the global variable `comparts`.